### DISCUSSION TITLE
Resource nextdns_parental_control failing

### DISCUSSION THREAD

**Discussion by przemekgorzynski** [amalucelli/terraform-provider-nextdns#23] [Q&A] [answered]
Hey,

While trying to deploy nextdns_parental_control resource I am going into error. Error says 

> Error: error updating parental control settings: %!v(PANIC=Format method: runtime error: invalid memory address or nil pointer dereference)
> with nextdns_parental_control.parental,
>  on nextdns_config.tf line 40, in resource "nextdns_parental_control" "parental":
> 40: resource "nextdns_parental_control" "parental" {

My code of block looks like below.

```
resource "nextdns_parental_control" "parental" {
  profile_id              = nextdns_profile.profile.id
  safe_search             = var.safe_search
  youtube_restricted_mode = var.youtube_restricted_mode
  block_bypass            = var.block_bypass
  
  service {
    id         = "tiktok"
    active     = true
    recreation = false
  }

  service {
    id         = "instagram"
    active     = false
    recreation = false
  }
}
```
Terraform plan gives correct output:

 

```
 # nextdns_parental_control.parental will be updated in-place
  ~ resource "nextdns_parental_control" "parental" {
        id                      = "333b9e"
        # (4 unchanged attributes hidden)

      + service {
          + active     = false
          + id         = "instagram"
          + recreation = false
        }
      + service {
          + active     = true
          + id         = "tiktok"
          + recreation = false
        }

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```
But apply still failing. Deployment of any other resources works fine, but this could not be deployed. 
Does anyone had the same error ?

**Comment by amalucelli** [comment, marked as answer]
This is a known issue, it was reported some time back and I even added some details to the thread: https://help.nextdns.io/t/35yyscy.
